packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -54,6 +54,13 @@ export interface DeploymentCircuitBreaker {
    */
   readonly rollback?: boolean;
 
+  /**
+   * Whether to use the explicit DeploymentControllerType.ECS when defining the circuit breaker.
+   * To avoid a service replacement when adding a CircuitBreaker to an existing ECS service, set this to false.
+   * @default true
+   */
+  readonly useExplicitEcsDeploymentController?: boolean;
```

Comment:
@fergusdixon I've been thinking about this and as much as a don't want to introduce a new feature flag I think that's the best thing to do here. We **want** the new default behavior to be `false` and a feature flag is the only way to accomplish that. 

https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#feature-flags

Reply:
Sure will give it a bash next week, thanks!

Reply:
@corymhall I have a working implementation with the flag, however I'm struggling to see this to `false` in the integration tests specifically. Should the integration tests cover new or old behaviour (I would imagine both), however I cant seem to set the feature flag value the same way as the unit tests do?

Reply:
It seems the [integ-helper sets all feature flags](https://github.com/fergusdixon/aws-cdk/blob/9e508fdbc8ec7e5b199a342bc7380641a0a0951c/tools/@aws-cdk/cdk-integ-tools/lib/integ-helpers.ts#L382) to `FUTURE_FLAGS` (new behaviour), and this takes precedence over the `context` prop passed to `App` in the test. So from this it seems integration tests should test new behaviour and unit tests for both?

Reply:
If you want to test the old behavior you can use the [postCliContext](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.AppProps.html#postclicontext) parameter on `App`.

Reply:
Have done, worked a charm, thanks!

Reply:
I really want to ship this, but this property is giving me pause. I really don't see the need for it.

* If you want the ECS controller explicitly, you can just supply it ("The customer is always right", below)
* If you do NOT want the ECS controller, you can set the feature flag 
  * If you want the ECS controller for *some* services and not other, you can either set the feature flag globally and add the explicit parameter to *some* services, or set the feature flag locally (you can flip a flag at any construct scope, it's all just context which is freely manipulable).

I think we can do without this, no?

Other than that, this is great!

Reply:
Just to cross-check "the need for the 'correct' default after this PR is merged": Now with current CDKv2 versions today, CFN output always has an ECS DeploymentController defined, if you have an CircuitBreaker enabled (even if you not aware as CDK user).
Once a stack has an ECS DeploymentController definied (= all stack build by CDK with CircuitBreaker enabled), you are not able to simply "remove this CFN-lines from JSON output" - If a new/upcoming CDK version would start to _not_ output ECS DeploymentController (by default) to JSON, all existing Services of existing Stacks would run into a replacement - Just because you update your CDK version.

As I understood the current code from Fergus, this default = true would remain the same behaviour as older CDKv2 version, right? - From my point of view that, it's an important brick, isn't it?